Person.rb

```diff
@@ -1,20 +1,30 @@
 require_relative 'nameable'
 
 class Person < Nameable
-  attr_accessor :id, :name, :age, :parent_permission
+  attr_accessor :id, :name, :age, :parent_permission, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   def can_use_services?
     of_age? >= 18 || @parent_permission
   end
 
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```

Comment:
- [OPTIONAL] Great job 👏🏻 But the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association eg:

```ruby
 def add_rental(book, date)
    Rental.new(date, book, self)
 end
```